app/src/endpoints.c

```diff
@@ -118,6 +132,24 @@ int zmk_endpoints_toggle_transport(void) {
 
 struct zmk_endpoint_instance zmk_endpoints_selected(void) { return current_instance; }
 
+struct zmk_endpoint_instance zmk_endpoints_preferred(void) {
```

Comment:
This looks like mostly duplicated code with `get_selected_instance()`. We should pull the common logic out into a separate function that both things can call so we don't keep the two functions in sync.

Reply:
I've added a new function `endpoints_add_extra_data()` which takes the transport (whether selected or preferred), adds the ble profile on if applicable and returns the full endpoint instance